src/interface/TWBitcoinCashAddress.cpp

```diff
@@ -28,33 +29,19 @@ bool TWBitcoinCashAddressIsValid(TWData *_Nonnull data) {
 }
 
 bool TWBitcoinCashAddressIsValidString(TWString *_Nonnull string) {
-    uint8_t data[104];
-    char hrpBuf[29];
-    size_t dataLen;
-    if (cash_decode(hrpBuf, data, &dataLen, TWStringUTF8Bytes(string)) == 0) {
-        return false;
-    }
-    if (strcmp(hrpBuf, hrp) != 0 || dataLen != dataSize) {
-        return false;
-    }
-    return true;
+    auto& stdString = *reinterpret_cast<const std::string*>(string);
+    return TW::Bitcoin::CashAddress::isValid(stdString);
 }
 
 bool TWBitcoinCashAddressInitWithString(struct TWBitcoinCashAddress *_Nonnull address, TWString *_Nonnull string) {
-    uint8_t data[104];
-    char hrpBuf[29];
-    size_t dataLen;
-    if (cash_decode(hrpBuf, data, &dataLen, TWStringUTF8Bytes(string)) == 0) {
+    auto& stdString = *reinterpret_cast<const std::string*>(string);
+    try {
+        const auto addr = TW::Bitcoin::CashAddress(stdString);
```

Comment:
Add test on which address it was failing?

Reply:
done